### ISSUE TITLE
Change directory error on mainline signing script

### ISSUE THREAD

**Issue by berglh** [berglh/ubuntu-sb-kernel-signing#1]
From u/FloatyFish,

Ref: [Reddit Comment](https://www.reddit.com/r/Ubuntu/comments/qjd1lq/comment/hno7wj4/?utm_source=share&utm_medium=web2x&context=3)

Below is what I get when I try to sign it:

```
/etc/kernel/postinst.d/00-mainline-signing: line 55: cd: too many arguments
run-parts: /etc/kernel/postinst.d/00-mainline-signing exited with return code 1
```

So I ran echo $HOME, and there weren't any spaces in the username folder. As for the kernel I'm trying to install, I was attempting to install 5.15.6 on Ubuntu 21.10.

What's even stranger is that I looked at upgrading my kernel (5.13.something) to the latest 5.14 release (5.14.21) to see if maybe it was a version issue, and to my surprise the build went through perfectly. I then attempted to build 5.15.6 and 5.15.0 to see if something had changed, and I still got the error that I posted in my previous post.

**Comment by berglh** [comment]
I tried this an 5.16.6 is working, small bug for 5.15.0.

**Comment by berglh** [comment]
After debugging, we found that it's possible for find to find multiple copies of the same deb. Even though we're checking for a response with "mainline" in the path, we're not check to see if there is more than one matching file. This should be fixed to ensure we only return lines from find that container mainline.

```
Mainline folder: /home/myname/.cache/mainline/5.15.6/amd64

/home/myname/.local/share/Trash/files, Arch: amd64, Kernel main version: v5.15.6

/etc/kernel/postinst.d/00-mainline-signing: line 56: cd: too many arguments
```

**Comment by berglh** [comment]
This has been solved in a2f400d
